pkg/fleet/env/env.go

```diff
@@ -108,13 +129,38 @@ func (e *Env) ToEnv() []string {
 	if e.RegistryAuthOverride != "" {
 		env = append(env, envRegistryAuth+"="+e.RegistryAuthOverride)
 	}
+	if len(e.ApmLibraries) > 0 {
+		libraries := []string{}
+		for l, v := range e.ApmLibraries {
+			l := string(l)
+			if v != "" {
+				l = l + ":" + string(v)
+			}
+			libraries = append(libraries, l)
+		}
+		slices.Sort(libraries)
+		env = append(env, envApmLibraries+"="+strings.Join(libraries, ","))
+	}
 	env = append(env, overridesByNameToEnv(envRegistryURL, e.RegistryOverrideByImage)...)
 	env = append(env, overridesByNameToEnv(envRegistryAuth, e.RegistryAuthOverrideByImage)...)
 	env = append(env, overridesByNameToEnv(envDefaultPackageInstall, e.DefaultPackagesInstallOverride)...)
 	env = append(env, overridesByNameToEnv(envDefaultPackageVersion, e.DefaultPackagesVersionOverride)...)
 	return env
 }
 
+func parseApmLibrariesEnv() map[ApmLibLanguage]ApmLibVersion {
+	apmLibraries := os.Getenv(envApmLibraries)
+	apmLibrariesVersion := map[ApmLibLanguage]ApmLibVersion{}
+	rest := apmLibraries
+	for rest != "" {
+		var library string
+		library, rest, _ = strings.Cut(rest, ",")
+		libraryName, libraryVersion, _ := strings.Cut(library, ":")
+		apmLibrariesVersion[ApmLibLanguage(libraryName)] = ApmLibVersion(libraryVersion)
+	}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Why not use `strings.Split`?

```suggestion
	pinnedLibraries := strings.Split(apmLibraries, ",")
	for _,  library := range pinnedLibraries {
		libraryName, libraryVersion, _ := strings.Cut(library, ":")
		apmLibrariesVersion[ApmLibLanguage(libraryName)] = ApmLibVersion(libraryVersion)
	}
```